### PR TITLE
Replace big.Int with constant-time safenum.Nat

### PR DESCRIPTION
**Pull request by marinthiercelin**
This pull requests replaces the use of big.Int with safenum.Nat to prevent the use of side-channel attacks to extract secrets from the execution of integer operations.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -3,8 +3,25 @@ github.com/ProtonMail/bcrypt v0.0.0-20210511135022-227b4adcab57 h1:pHA4K54ifoogV
...
+github.com/cronokirby/safenum v0.11.0 h1:XD9ikQk+SuCwQ4LuYllHCQd2/LBUIEx/2QH3pckbMMo=
+github.com/cronokirby/safenum v0.11.0/go.mod h1:AWp82xwEqKcnrpJPXPa1m0gF/OY8dzgL17ubUBnVygA=
+github.com/cronokirby/safenum v0.13.0 h1:1TAQXC30E9RWmjKwU7LgywsMRJkA6UgL+vZaU6NJegQ=
+github.com/cronokirby/safenum v0.13.0/go.mod h1:AWp82xwEqKcnrpJPXPa1m0gF/OY8dzgL17ubUBnVygA=
+github.com/cronokirby/safenum v0.14.0 h1:etZ9FnkWojRAphX2PeOquGX8Um6NqE02fHwLi9Xswx0=
+github.com/cronokirby/safenum v0.14.0/go.mod h1:AWp82xwEqKcnrpJPXPa1m0gF/OY8dzgL17ubUBnVygA=
+github.com/cronokirby/safenum v0.15.0 h1:WnAWW4lpQkPSQIXvyUZe99iaR5IBGPA/eQmCdU2EEN4=
+github.com/cronokirby/safenum v0.15.0/go.mod h1:AWp82xwEqKcnrpJPXPa1m0gF/OY8dzgL17ubUBnVygA=
+github.com/cronokirby/safenum v0.25.0 h1:3owHW181y3Ns+7nMRH0W9JuLi1H63imy6EngBlh9VHQ=
+github.com/cronokirby/safenum v0.25.0/go.mod h1:AWp82xwEqKcnrpJPXPa1m0gF/OY8dzgL17ubUBnVygA=
```
Comment (twiss) [inline go.sum:16]: This is very nitpicky, and idk if there's a command for this, but maybe let's clean this up a bit
```suggestion
```

File: version.go [hunk 2]
```diff
@@ -22,7 +22,9 @@
 
 package srp
 
+const Version string = "0.0.0"
```
Comment (twiss) [inline version.go:25]: This also doesn't much matter, but maybe let's say that this is 0.0.1, and that the previous "undefined"(?!) version was 0.0.0, or something..
```suggestion
const Version string = "0.0.1"
```

**Review by twiss — COMMENTED** [review]
LGTM :+1: Some inconsequential nitpicks:

File: go.sum (hunk 1 above)
Comment (marinthiercelin) [reply go.sum]: `go mod tidy` did it, thanks

**Review by twiss — APPROVED** [review]

**Pull request merged**